masonry/src/text/selection.rs

```diff
@@ -283,6 +284,120 @@ impl<T: Selectable> TextWithSelection<T> {
         }
         self.layout.draw(scene, point);
     }
+
+    fn access_position_from_offset(
+        &self,
+        parent_node: &NodeBuilder,
+        offset: usize,
+        affinity: Affinity,
+    ) -> TextPosition {
+        let text = self.text().as_ref();
+        assert!(offset <= text.len());
+        if offset == text.len() {
+            let last_id = *parent_node.children().last().unwrap();
+            let character_lengths = self
+                .layout
+                .character_lengths_by_access_id
+                .get(&last_id)
+                .unwrap();
+            return TextPosition {
+                node: last_id,
+                character_index: character_lengths.len(),
+            };
+        }
+        for (line_index, line) in self.layout.layout.lines().enumerate() {
```

Comment:
I think we can early-continue the search if the line can't contain the given offset.